webapp/src/components/TransactionDetails.test.tsx

```diff
@@ -0,0 +1,51 @@
+import "@testing-library/jest-dom";
+import { render, screen } from "@testing-library/react";
+import { describe, expect, it } from "vitest";
+import { TransactionDetails } from "./TransactionDetails";
+
+describe("TransactionDetails", () => {
+	const baseDetails = {
+		to: "0x123",
+		value: "1000",
+		data: "0xdeadbeef",
+		operation: 0,
+		safeTxGas: "0",
+		baseGas: "0",
+		gasPrice: "0",
+		gasToken: "0x0",
+		refundReceiver: "0x0",
+		stored: true,
+	};
+
+	it("renders all fields and decode link for non-empty data", () => {
+		render(<TransactionDetails details={baseDetails} />);
+		// The param names are in a <strong> tag, so we need to get the parent element.
+		expect(screen.getByText(/To:/).parentElement).toHaveTextContent("0x123");
```

Comment:
I guess this is one of the downsides to UI tests, is that they are sensitive to page layout (so refactoring the display can potentially cause a lot of test refactors as well).